packages/frontend/core/src/components/pure/swr-error-bundary.tsx

```diff
@@ -0,0 +1,56 @@
+import type { ErrorInfo } from 'react';
+import React, { useRef } from 'react';
+import type { ErrorBoundaryProps } from 'react-error-boundary';
+import { ErrorBoundary } from 'react-error-boundary';
+import { useSWRConfig } from 'swr';
+
+/**
+ * If we use suspense mode in SWR, we need to preload or delete cache to retry request.
+ * Or the error will be cached and the request will not be retried.
+ *
+ * Reference:
+ * https://github.com/vercel/swr/issues/2740
+ * https://github.com/vercel/swr/blob/main/core/src/use-swr.ts#L690
+ * https://github.com/vercel/swr/tree/main/examples/suspense-retry
+ */
+export const SWRErrorBoundary = (props: ErrorBoundaryProps) => {
+  const { onReset, onError } = props;
+  const errorsRef = useRef<Error[]>([]);
+  const { cache } = useSWRConfig();
+
+  const clearErrorCache = React.useCallback(() => {
+    const errors = errorsRef.current;
+    errorsRef.current = [];
+
+    for (const key of cache.keys()) {
+      const item = cache.get(key);
+      if (errors.includes(item?.error)) {
+        cache.delete(key);
+      }
+    }
+  }, [cache]);
+
+  const onResetWithSWR = React.useCallback(
+    (details: any) => {
+      clearErrorCache();
+      onReset?.(details);
+    },
+    [clearErrorCache, onReset]
+  );
+
+  const onErrorWithSWR = React.useCallback(
+    (error: Error, info: ErrorInfo) => {
+      errorsRef.current.push(error);
+      onError?.(error, info);
+    },
+    [onError]
+  );
+
+  React.useEffect(() => clearErrorCache, [clearErrorCache]);
```

Comment:
I've added this logic to ensure that resend request when component reloaded(e.g. switch tab and back). @joooye34

Reply:
ok...that Is cool

Reply:
Why is this function not executed